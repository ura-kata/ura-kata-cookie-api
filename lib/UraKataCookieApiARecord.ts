import { ARecord, RecordTarget, IHostedZone } from '@aws-cdk/aws-route53';
import { Construct } from '@aws-cdk/core';
import { ApiGatewayDomain } from '@aws-cdk/aws-route53-targets';
import { UraKataCookieApiCustomDomainName } from './UraKataCookieApiCustomDomainName';

export class UraKataCookieApiARecord extends ARecord {
  constructor(
    scope: Construct,
    id: string,
    customDomainName: UraKataCookieApiCustomDomainName,
    hostedZone: IHostedZone
  ) {
    super(scope, id, {
      target: RecordTarget.fromAlias(new ApiGatewayDomain(customDomainName)),
      recordName: customDomainName.domainName,
      zone: hostedZone,
    });
  }
}
