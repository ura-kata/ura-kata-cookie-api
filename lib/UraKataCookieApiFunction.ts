import { NodejsFunction } from '@aws-cdk/aws-lambda-nodejs';
import { Construct } from '@aws-cdk/core';
import { Runtime } from '@aws-cdk/aws-lambda';

export class UraKataCookieApiFunction extends NodejsFunction {
  constructor(
    scope: Construct,
    id: string,
    functionEntry: string,
    functionName: string,
    environment?: { [key: string]: string }
  ) {
    super(scope, id, {
      entry: functionEntry,
      functionName: functionName,
      runtime: Runtime.NODEJS_14_X,
      environment: environment,
    });
  }
}
