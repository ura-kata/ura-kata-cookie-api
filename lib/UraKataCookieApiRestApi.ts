import { RestApi } from '@aws-cdk/aws-apigateway';
import { Construct } from '@aws-cdk/core';

export class UraKataCookieApiRestApi extends RestApi {
  constructor(scope: Construct, id: string, apiName: string) {
    super(scope, id, {
      restApiName: apiName,
    });
  }
}
