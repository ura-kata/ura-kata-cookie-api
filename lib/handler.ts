import { APIGatewayProxyHandler, APIGatewayProxyResult } from 'aws-lambda';

export const handler: APIGatewayProxyHandler = async (
  event,
  context
): Promise<APIGatewayProxyResult> => {
  const statusCode = 200;
  const multiValueHeaders = {};
  const responseBody = JSON.stringify(event);
  let responseHeaders = {
    'Content-Type': 'application/json',
  };

  console.log(event);
  return {
    statusCode: statusCode,
    body: responseBody,
    headers: responseHeaders,
    multiValueHeaders,
  };
};
