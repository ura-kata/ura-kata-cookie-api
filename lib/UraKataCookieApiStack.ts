import { LambdaIntegration } from '@aws-cdk/aws-apigateway';
import * as cdk from '@aws-cdk/core';
import * as path from 'path';
import { UraKataCookieApiARecord } from './UraKataCookieApiARecord';
import { UraKataCookieApiCustomDomainName } from './UraKataCookieApiCustomDomainName';
import { UraKataCookieApiFunction } from './UraKataCookieApiFunction';
import { UraKataCookieApiRestApi } from './UraKataCookieApiRestApi';
import { HostedZone } from '@aws-cdk/aws-route53';
import * as dotenv from 'dotenv';
import { Certificate } from '@aws-cdk/aws-certificatemanager';

dotenv.config();

export class UraKataCookieApiStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    const zoneName = process.env.URA_KATA_COOKIE_API_DOMAIN_NAME as string;
    const domainName = `${process.env.URA_KATA_COOKIE_API_HOST_NAME}.${process.env.URA_KATA_COOKIE_API_DOMAIN_NAME}`;
    const certificateArn = process.env.URA_KATA_CERTIFICATE_ARN as string;
    const publicHostedZoneId = process.env
      .URA_KATA_PUBLIC_HOSTED_ZONE_ID as string;

    const hostZone = HostedZone.fromHostedZoneAttributes(
      this,
      'UraKataPublicHostedZone',
      {
        hostedZoneId: publicHostedZoneId,
        zoneName: zoneName,
      }
    );

    const certificate = Certificate.fromCertificateArn(
      this,
      'UraKataCertificate',
      certificateArn
    );

    const customDomainName = new UraKataCookieApiCustomDomainName(
      this,
      'UraKataCookieApiCustomDomainName',
      domainName,
      certificate
    );

    new UraKataCookieApiARecord(
      this,
      'UraKataCookieApiARecord',
      customDomainName,
      hostZone,
      domainName
    );

    const lambdaFunction = new UraKataCookieApiFunction(
      this,
      'UraKataCookieApiLambda',
      path.join(__dirname, '../lib/handler.ts'),
      'ura-kata-cookie-api'
    );

    const restApi = new UraKataCookieApiRestApi(
      this,
      'UraKataCookieApiRestApiGateway',
      'ura-kata-cookie-api'
    );

    const integration = new LambdaIntegration(lambdaFunction);

    const setResouse = restApi.root.addResource('set');
    const resetResouse = restApi.root.addResource('reset');

    setResouse.addMethod('POST', integration);
    resetResouse.addMethod('POST', integration);
  }
}
