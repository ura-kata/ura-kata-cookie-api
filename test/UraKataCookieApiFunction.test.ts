import { Stack } from '@aws-cdk/core';
import { UraKataCookieApiFunction } from '../lib/UraKataCookieApiFunction';
import * as path from 'path';
import { SynthUtils } from '@aws-cdk/assert';

test('ura-kata-cookie-api-function creates', () => {
  const stack = new Stack();
  new UraKataCookieApiFunction(
    stack,
    'func',
    path.join(__dirname, '../lib/handler.ts'),
    'func-name'
  );
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
