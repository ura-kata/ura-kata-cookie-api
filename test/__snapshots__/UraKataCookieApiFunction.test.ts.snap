// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ura-kata-cookie-api-function creates 1`] = `
Object {
  "Parameters": Object {
    "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6ArtifactHash6BF79A87": Object {
      "Description": "Artifact hash for asset \\"0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6\\"",
      "Type": "String",
    },
    "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6S3Bucket615074D3": Object {
      "Description": "S3 bucket for asset \\"0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6\\"",
      "Type": "String",
    },
    "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6S3VersionKey6BC05444": Object {
      "Description": "S3 key for asset version \\"0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6\\"",
      "Type": "String",
    },
  },
  "Resources": Object {
    "funcC3A0C2E2": Object {
      "DependsOn": Array [
        "funcServiceRoleA96CCB44",
      ],
      "Properties": Object {
        "Code": Object {
          "S3Bucket": Object {
            "Ref": "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6S3Bucket615074D3",
          },
          "S3Key": Object {
            "Fn::Join": Array [
              "",
              Array [
                Object {
                  "Fn::Select": Array [
                    0,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6S3VersionKey6BC05444",
                        },
                      ],
                    },
                  ],
                },
                Object {
                  "Fn::Select": Array [
                    1,
                    Object {
                      "Fn::Split": Array [
                        "||",
                        Object {
                          "Ref": "AssetParameters0f9ada343598c77e5f53cc2cb8fea79de87331c376ee1b2555824ba97be34ee6S3VersionKey6BC05444",
                        },
                      ],
                    },
                  ],
                },
              ],
            ],
          },
        },
        "Environment": Object {
          "Variables": Object {
            "AWS_NODEJS_CONNECTION_REUSE_ENABLED": "1",
          },
        },
        "FunctionName": "func-name",
        "Handler": "index.handler",
        "Role": Object {
          "Fn::GetAtt": Array [
            "funcServiceRoleA96CCB44",
            "Arn",
          ],
        },
        "Runtime": "nodejs14.x",
      },
      "Type": "AWS::Lambda::Function",
    },
    "funcServiceRoleA96CCB44": Object {
      "Properties": Object {
        "AssumeRolePolicyDocument": Object {
          "Statement": Array [
            Object {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": Object {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": Array [
          Object {
            "Fn::Join": Array [
              "",
              Array [
                "arn:",
                Object {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
}
`;
